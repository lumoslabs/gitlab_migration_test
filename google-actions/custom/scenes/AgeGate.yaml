intentEvents:
- handler:
    webhookHandler: AgeGateResult
  intent: AGE_GATE_RESULT
