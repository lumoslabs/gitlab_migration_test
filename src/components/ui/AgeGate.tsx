import React, { useState } from 'react'
import DatePicker from 'react-mobile-datepicker'
import { StyleSheet, css } from 'aphrodite'
import { Container, Row } from 'react-bootstrap'
import base from '@styles/colors/base'
import commonStyles from '@styles/commonStyles'
import Button from '@components/ui/Button'

export interface IAgeGateProps {
  onSubmit(birthday: Date): any;
  max?: Date,
  min?: Date,
  disabled?: boolean
}

const AgeGate = (({
  onSubmit,
  max = (new Date(Date.now() - 86400000)),
  min = (new Date('1900-01-02')),
  disabled = false
}: IAgeGateProps): JSX.Element => {

  const [date, setDate] = useState(new Date('1985-01-02'))

  const handleClick = () => {
    //    const isUnderage = !(new Date(Date.now()).getTime() - new Date(date).getTime() > 410310589331)
    onSubmit(date)
  }

  const monthMap = {
    '1': 'Jan',
    '2': 'Feb',
    '3': 'Mar',
    '4': 'Apr',
    '5': 'May',
    '6': 'Jun',
    '7': 'Jul',
    '8': 'Aug',
    '9': 'Sep',
    '10': 'Oct',
    '11': 'Nov',
    '12': 'Dec',
  }

  const dateConfig = {
    'month': {
      format: value => monthMap[value.getMonth() + 1],
      caption: 'Mon',
      step: 1,
    },
    'date': {
      format: 'DD',
      caption: 'Day',
      step: 1,
    },
    'year': {
      format: 'YYYY',
      caption: 'Year',
      step: 1,
    },
  }

  return (
    <Container className={css(commonStyles.flexColumnAlignCenter)}>
      <Row className={css(commonStyles.flexRowAllCenter)}>
        <p className={css(styles.title)}>
          {'Please select your birthdate:'}
        </p>
      </Row>
      <Row className={css(commonStyles.flexRowAllCenter, styles.datePickerRow)}>
        <DatePicker
          theme='android'
          dateConfig={dateConfig}
          showHeader={false}
          showFooter={false}
          value={date}
          onChange={(date) => { setDate(new Date(date)) }}
          isOpen={true}
          isPopup={false}
          max={max}
          min={min}
        />
      </Row>
      {/* TODO: Update with legal approved copy */}
      <p className={css(styles.disclaimer)}>
          {'Your birthday will be stored and added to your Lumosity account if you choose to create one.'}
      </p>
      <Row className={css(commonStyles.flexRowAllCenter, styles.buttonDiv)}>
        <Button
          disabled={disabled}
          onClick={handleClick}
          text='Submit'
          eventData={{ id: 'submit_birthdate' }}
        />
      </Row>
    </Container>
  )
})

export default AgeGate

const styles = StyleSheet.create({
  title: {
    color: base.lumosBlack,
    fontFamily: 'MuseoSans500',
    fontSize: '32px',
    fontWeight: 700,
    padding: '0px',
    marginTop: '8vh',
    textAlign: 'center'
  },
  disclaimer: {
    color: base.lumosBlack,
    fontFamily: 'MuseoSans500',
    fontSize: '24px',
    fontWeight: 500,
    padding: '0px',
    textAlign: 'center'
  },
  buttonDiv: {
    bottom: '12vmin'
  },
  datePickerRow: {
    height: '50vh'
  }
})
