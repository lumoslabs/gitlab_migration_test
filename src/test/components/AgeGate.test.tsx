import React from 'react'
import AgeGate from '@components/ui/AgeGate'
import { render } from '@test/render-with-redux'

const onSubmit = () => { window.location.href = '/' }

describe('components', () => {
  describe('AgeGate', () => {
    it('matches snapshot', () => {
      const { asFragment } = render(
        <AgeGate
          max={new Date(Date.UTC(2021, 1, 1, 1, 1, 0, 0))}
          min={new Date(Date.UTC(1900, 1, 1, 1, 1, 0, 0))}
          onSubmit={onSubmit}
        />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
