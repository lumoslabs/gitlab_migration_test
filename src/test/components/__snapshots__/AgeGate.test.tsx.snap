// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flexColumnAlignCenter_1ndpwxc container"
  >
    <div
      class="flexRowAllCenter_1cn94ih row"
    >
      <p
        class="title_8noqbn"
      >
        Please select your birthdate:
      </p>
    </div>
    <div
      class="flexRowAllCenter_1cn94ih-o_O-datePickerRow_1dmvaf1 row"
    >
      <div
        class="datepicker android"
      >
        <div
          class="datepicker-content"
        >
          <div
            class="datepicker-col-1"
          >
            <div
              class="datepicker-viewport"
            >
              <div
                class="datepicker-wheel"
              >
                <ul
                  class="datepicker-scroll"
                  style="transform: translateY(-200px); margin-top: 0px;"
                >
                  <li
                    class=""
                  >
                    Sep
                  </li>
                  <li
                    class=""
                  >
                    Oct
                  </li>
                  <li
                    class=""
                  >
                    Nov
                  </li>
                  <li
                    class=""
                  >
                    Dec
                  </li>
                  <li
                    class=""
                  >
                    Jan
                  </li>
                  <li
                    class=""
                  >
                    Feb
                  </li>
                  <li
                    class=""
                  >
                    Mar
                  </li>
                  <li
                    class=""
                  >
                    Apr
                  </li>
                  <li
                    class=""
                  >
                    May
                  </li>
                  <li
                    class=""
                  >
                    Jun
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="datepicker-col-1"
          >
            <div
              class="datepicker-viewport"
            >
              <div
                class="datepicker-wheel"
              >
                <ul
                  class="datepicker-scroll"
                  style="transform: translateY(-200px); margin-top: 0px;"
                >
                  <li
                    class=""
                  >
                    27
                  </li>
                  <li
                    class=""
                  >
                    28
                  </li>
                  <li
                    class=""
                  >
                    29
                  </li>
                  <li
                    class=""
                  >
                    30
                  </li>
                  <li
                    class=""
                  >
                    31
                  </li>
                  <li
                    class=""
                  >
                    01
                  </li>
                  <li
                    class=""
                  >
                    02
                  </li>
                  <li
                    class=""
                  >
                    03
                  </li>
                  <li
                    class=""
                  >
                    04
                  </li>
                  <li
                    class=""
                  >
                    05
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="datepicker-col-1"
          >
            <div
              class="datepicker-viewport"
            >
              <div
                class="datepicker-wheel"
              >
                <ul
                  class="datepicker-scroll"
                  style="transform: translateY(-200px); margin-top: 0px;"
                >
                  <li
                    class=""
                  >
                    1980
                  </li>
                  <li
                    class=""
                  >
                    1981
                  </li>
                  <li
                    class=""
                  >
                    1982
                  </li>
                  <li
                    class=""
                  >
                    1983
                  </li>
                  <li
                    class=""
                  >
                    1984
                  </li>
                  <li
                    class=""
                  >
                    1985
                  </li>
                  <li
                    class=""
                  >
                    1986
                  </li>
                  <li
                    class=""
                  >
                    1987
                  </li>
                  <li
                    class=""
                  >
                    1988
                  </li>
                  <li
                    class=""
                  >
                    1989
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <p
      class="disclaimer_kdqsn7"
    >
      Your birthday will be stored and added to your Lumosity account if you choose to create one.
    </p>
    <div
      class="flexRowAllCenter_1cn94ih-o_O-buttonDiv_120lt8p row"
    >
      <button
        class="button_15gg8qi"
      >
        Submit
      </button>
    </div>
  </div>
</DocumentFragment>
`;
